### DISCUSSION TITLE
需要一个更多读弹幕的音色

### DISCUSSION THREAD

**Discussion by xiaoyekong2** [xfgryujk/blivechat#206] [想法]
现有的插件是通过微软api做到的读弹幕，音色过于机械，大佬们可以开发一个有有道或百度的音色，或者自定义音色api的插件

**Comment by xfgryujk** [comment]
这个功能不是为了好玩而做的，所以我只是用了最简单可靠的方案。如果用远程的TTS服务，要考虑网络问题，而且可能要计费，配置起来比较麻烦。如果本地跑AI合成语音，配置麻烦，模型占用空间大，还有计算开销，这是不可接受的

**Comment by AoEiuV020** [comment]
考虑无鉴权的免费api？哪怕不够可靠也比没有强吧，加个初始化成功的播报和失败的弹幕就就当解决播报不稳定问题了，

https://dds.dui.ai/runtime/v1/synthesize?voiceId=kaolam_diantai&text=%E6%88%91%E4%BB%AC%E9%9C%80%E8%A6%81%E8%AF%AD%E9%9F%B3%E6%92%AD%E6%8A%A5%E5%8A%9F%E8%83%BD&speed=1&volume=100&audioType=wav

https://czyt.tech/post/a-free-tts-api/

**Comment by AoEiuV020** [comment]
考虑Web Speech API， 前端js直接调用， 系统提供音色播报，免费无限制，完全可以直接集成到本体里，没插件时兜底使用， 
```js
window.speechSynthesis.speak(new SpeechSynthesisUtterance("测试语音播报"));
```
这有个demo， https://eeejay.github.io/webspeechdemos/
mac自带音色还挺多， 
![image](https://github.com/user-attachments/assets/47416f55-0b27-4a19-8636-78ac071905b5)

**Reply by xfgryujk** [reply]
这个我早知道了，也就比现在的插件多了谷歌的一些在线语音

**Reply by AoEiuV020** [reply]
> 这个我早知道了，也就比现在的插件多了谷歌的一些在线语音

主要感觉这个没什么消耗可以不用做成插件， 直接支持， 这样非windows也能用， 而且能在公共服务器使用， 省的自己搭本地服务器了，

**Reply by xfgryujk** [reply]
> > 这个我早知道了，也就比现在的插件多了谷歌的一些在线语音
> 
> 主要感觉这个没什么消耗可以不用做成插件， 直接支持， 这样非windows也能用， 而且能在公共服务器使用， 省的自己搭本地服务器了，

就是不想放到核心功能里才做成插件的
